### ISSUE TITLE
Design DB Layout

### ISSUE THREAD

**Issue by Zedjones** [Zedjones/PokePredict#1]
Some things to consider:
 - What information do we need to store?
 - What database do we use?
 - How does the PokeAPI factor into all of this?
 - How will Pokemon, moves, etc. relate to each other in the DB?

**Comment by Zedjones** [comment]
At this point, I've come up with the database layout and how to store the data.
Given that we only have ~1000 static entries with a limited amount of information totaling around 30 MB, I've gone with the simplest solution which I believe makes sense given the constraints:
 - Serializing two types of objects as JSON: Pokemon and types
   - Each Pokemon contains a list of moves, a string correlating to the type, base stats, the Pokemon's name, and weight
   - Each move contains the power, name, string correlating to the type, priority, a list of stat changes, and the target of the move (user, enemy, etc.)
   - Each type contains the list of damage relations (damage to and damage from) and the type name
     - Each damage relation contains the damage multiplier, the target type, and target type name

**Comment by Zedjones** [comment]
Using the previous approach, I ended up crashing the runtime while trying to simultaneously fetch all of the Pokemon from the API. In addition, even when it wouldn't crash, fetching one Pokemon synchronously would take an upwards of 30 seconds. 

Since I didn't want to try and re-engineer the API solution I had already come up with and couldn't think of a way to do it any more efficiently, I decided to use the database that was backing the API: [Veekun](https://github.com/veekun/pokedex). Using the instructions included in their Wiki, I was able to generate a SQLite database containing the information that I needed. And using the Entity Framework's database-first approach (using `ef scaffold`), I was able to generate the models needed. This is the current approach being used in the `master` branch
